### PR TITLE
String validation

### LINKED ISSUE #53: String Validation
# Backend String Validation
- Validate all the string fields in the backend models. This includes phone numbers, emails, and so on.
- Update endpoints accordingly based on the validation

### PR DESCRIPTION
**Pull request by Leundai**
## Summary
closes #53 

Validates specified fields in the backend given a regex. 

It supports multiple variations of phone numbers. Whether they are spaced using '.' , '-', or space

## Changes

- Added regex testing for specified fields
- updated validateMemberQuery to support the regex

### CODE REVIEW

**Review by daniel-moon32 — APPROVED** [review]
Looks good!

File: api/src/utils/user-utils.js [hunk 1]
```diff
@@ -53,9 +60,17 @@ const filterViewableFields = (currentUser, member) => {
   return filteredMember;
 };
 
+const validateField = (field, value) => {
```
Comment (SirajChokshi) [inline api/src/utils/user-utils.js:63]: For this function lets consider decoupling it from our global state. I would be more modular to pass in the validation fields along with our other arguments.

File: api/src/utils/user-utils.js [hunk 2]
```diff
@@ -24,6 +24,13 @@ const nonEditableFields = [
   'role',
   'status',
 ];
+// Fields and their regex for validation
+const validationFields = {
+  email: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
+  phone: /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/,
```
Comment (SirajChokshi) [inline api/src/utils/user-utils.js:30]: This query allows for phone numbers in a variety of formats (978754-5343, 978.754-5343, etc). Should we instead decide to enforce a 10 digit sequence and splice out delimiters like spaces, hyphens, and `.`'s on the frontend? This way we can also enforce a standardized format as well as one that better guarantees legitimate phone numbers.

e.g. User enters `555-545-7565` on the frontend and it gets sent to the API as `5555457565` and validated as such

**Review by SirajChokshi — CHANGES_REQUESTED** [review]
Looks good, few comments :)

**Review by SirajChokshi — APPROVED** [review]
Great work!

**Pull request merged**